src/server/system_services/bucket_server.js

```diff
@@ -1142,7 +1142,23 @@ function get_cloud_buckets(req) {
                 return storage.getBuckets()
                     .then(data => data[0].map(bucket =>
                         _inject_usage_to_cloud_bucket(bucket.name, connection.endpoint, used_cloud_buckets)));
-            } else { //else if AWS
+            } else if (connection.endpoint_type === 'AWSSTS') {
+               const additionalParams = {
+                RoleSessionName: "get_cloud_buckets_session",
+                signatureVersion: cloud_utils.get_s3_endpoint_signature_ver(connection.endpoint, connection.auth_method),
+                s3DisableBodySigning: cloud_utils.disable_s3_compatible_bodysigning(connection.endpoint),
+                httpOptions: {
+                    agent: http_utils.get_unsecured_agent(connection.endpoint)
+                }
+               };
+               const s3 = await cloud_utils.createSTSS3Client(connection, additionalParams);
```

Comment:
As this is all the same code as in the else (correct me if wrong) I think also preferred to use generate_aws_sts_creds here as well and to split the s3 params between the options.

Reply:
Done.